test/unittesting/config_test.go

```diff
@@ -0,0 +1,1403 @@
+/*
+ * Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package unittesting
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/supertokens/supertokens-golang/recipe/session"
+	"github.com/supertokens/supertokens-golang/recipe/session/sessmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+func TestSuperTokensInitWithAbsentOptionalFieldsInAppInfo(t *testing.T) {
+	apiBasePath := "test/"
+	websiteBasePath := "test1/"
+	configValues := []supertokens.TypeInput{
+		{
+			Supertokens: &supertokens.ConnectionInfo{
+				ConnectionURI: "http://localhost:8080",
+			},
+			AppInfo: supertokens.AppInfo{
+				AppName:       "SuperTokens",
+				APIDomain:     "api.supertokens.io",
+				WebsiteDomain: "supertokens.io",
+			},
+			RecipeList: []supertokens.Recipe{
+				session.Init(nil),
+			},
+		},
+		{
+			Supertokens: &supertokens.ConnectionInfo{
+				ConnectionURI: "http://localhost:8080",
+			},
+			AppInfo: supertokens.AppInfo{
+				AppName:         "SuperTokens",
+				APIDomain:       "api.supertokens.io",
+				WebsiteDomain:   "supertokens.io",
+				APIBasePath:     &apiBasePath,
+				WebsiteBasePath: &websiteBasePath,
+			},
+			RecipeList: []supertokens.Recipe{
+				session.Init(nil),
+			},
+		},
+	}
+	for _, configValue := range configValues {
+
+		BeforeEach()
+		StartUpST("localhost", "8080")
+		err := supertokens.Init(configValue)
+		if err != nil {
+			t.Error("Failed to get a supertokens instance")
+		}
+		supertokensInstance, err := supertokens.GetInstanceOrThrowError()
+
+		if err != nil {
+			t.Error("could not find a supertokens instance")
+		}
+
+		if configValue.AppInfo.APIBasePath != nil {
+			assert.Equal(t, "/test", supertokensInstance.AppInfo.APIBasePath.GetAsStringDangerous())
+			assert.Equal(t, "/test1", supertokensInstance.AppInfo.WebsiteBasePath.GetAsStringDangerous())
+		} else {
+			assert.Equal(t, "/auth", supertokensInstance.AppInfo.APIBasePath.GetAsStringDangerous())
+			assert.Equal(t, "/auth", supertokensInstance.AppInfo.WebsiteBasePath.GetAsStringDangerous())
+		}
+		AfterEach()
+	}
```

Comment:
This type of testing is very weird.. You want to either separate them into different tests, or do init -> check -> init -> check.

Also, do you really need to start and stop ST in each loop here? Cause that will add more time to this test